addons/account/models/account_move.py

```diff
@@ -4084,7 +4084,7 @@ def action_open_business_doc(self):
         }
 
     def action_update_fpos_values(self):
-        self.invoice_line_ids._compute_price_unit()
+        self.invoice_line_ids.with_context(fpos_change=True)._compute_price_unit()
```

Comment:
Adding special behavior with context keys should be avoided as much as possible imo. Why not add the check on taxes here and prevent calling _compute_price_unit if not needed ?